postings.go

```diff
@@ -240,6 +241,40 @@ func (it *listPostings) Err() error {
 	return nil
 }
 
+type bytePostings struct {
+	list []byte
+	idx  int
+}
+
+func newBytePostings(list []byte) *bytePostings {
+	return &bytePostings{list: list, idx: -1}
+}
+
+func (it *bytePostings) At() uint32 {
+	idx := 4 * it.idx
+	return binary.BigEndian.Uint32(it.list[idx : idx+4])
+}
+
+func (it *bytePostings) Next() bool {
+	it.idx++
+	return it.idx*4 < len(it.list)
+}
+
+func (it *bytePostings) Seek(x uint32) bool {
+	num := len(it.list) / 4
+	// Do binary search between current position and end.
+	it.idx += sort.Search(num-it.idx, func(i int) bool {
+		idx := 4 * (it.idx + i)
+		val := binary.BigEndian.Uint32(it.list[idx : idx+4])
+		return val >= x
+	})
```

Comment:
As another little project, what would be interesting here is setting up some benchmarking around seeking for `x` if the current value is close to `x`.

If we seek for 10 and are currently at 8, initiating a binary search might be way more expensive than just streaming forward sequentially.
Aside from proper benchmarking, this probably also needs some understanding around how frequent that case is in the real world when intersecting postings for example.

At this point it's probably a bit premature to think about.